tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -4878,6 +4878,8 @@ void CGMSHLSLRuntime::EmitHLSLAggregateCopy(
       return;
     }
     const clang::RecordType *RT = Type->getAsStructureType();
+    if (!RT)
```

Comment:
Would be good to figure out which case getAsStructureType is missing, and either enlighten that side, or (more likely) have a version that does the correct type-chasing.